shared/services/wallet/masquerade-wallet.go

```diff
@@ -0,0 +1,246 @@
+}
+
+// End a masquerade, restoring your node's effective address back to your wallet address if one is loaded
+func (w *masqueradeWallet) EndMasquerade() error {
+	if w.am == nil {
+		return errors.New("node is not masquerading")
+	}
+
+	return w.am.DeleteAddressFile()
+}
+
+// Gets the wallet's chain ID
+func (w *masqueradeWallet) GetChainID() *big.Int {
+	copy := big.NewInt(0).Set(w.chainID)
+	return copy
+}
+
+// Add a keystore to the wallet
+func (w *masqueradeWallet) AddKeystore(name string, ks keystore.Keystore) {
+	return
+}
+
+// Always return true as we're masquerading
+func (w *masqueradeWallet) IsInitialized() bool {
+	return true
+}
+
+// Attempt to initialize the wallet if not initialized and return status
+func (w *masqueradeWallet) GetInitialized() (bool, error) {
+	if w.IsInitialized() {
+		return true, nil
+	}
+	return w.loadStore()
+}
+
+// Serialize the wallet to a JSON string
+func (w *masqueradeWallet) String() (string, error) {
+	return "", ErrIsMasquerading
+
+}
+
+// Initialize the wallet from a random seed
+func (w *masqueradeWallet) Initialize(derivationPath string, walletIndex uint) (string, error) {
+	return "", ErrIsMasquerading
+
+}
+
+// Recover a wallet from a mnemonic
+func (w *masqueradeWallet) Recover(derivationPath string, walletIndex uint, mnemonic string) error {
+	return ErrIsMasquerading
+
+}
+
+// Recover a wallet from a mnemonic - only used for testing mnemonics
+func (w *masqueradeWallet) TestRecovery(derivationPath string, walletIndex uint, mnemonic string) error {
+
+	// Check mnemonic
+	if !bip39.IsMnemonicValid(mnemonic) {
+		return fmt.Errorf("Invalid mnemonic '%s'", mnemonic)
+	}
+
+	// Generate seed
+	w.seed = bip39.NewSeed(mnemonic, "")
+
+	// Create master key
+	var err error
+	w.mk, err = hdkeychain.NewMaster(w.seed, &chaincfg.MainNetParams)
+	if err != nil {
+		return fmt.Errorf("Could not create wallet master key: %w", err)
+	}
+
+	// Create wallet store
+	w.ws = &walletStore{
+		Name:           w.encryptor.Name(),
+		Version:        w.encryptor.Version(),
+		UUID:           uuid.New(),
+		DerivationPath: derivationPath,
+		WalletIndex:    walletIndex,
+		NextAccount:    0,
+	}
+
+	// Return
+	return nil
+
+}
+
+// Save the wallet store to disk
+func (w *masqueradeWallet) Save() error {
+	return ErrIsMasquerading
+}
+
+// Delete the wallet store from disk
+func (w *masqueradeWallet) Delete() error {
+	return ErrIsMasquerading
+}
+
+// Signs a serialized TX using the wallet's private key
+func (w *masqueradeWallet) Sign(serializedTx []byte) ([]byte, error) {
+	return nil, ErrIsMasquerading
+}
+
+// Signs an arbitrary message using the wallet's private key
+func (w *masqueradeWallet) SignMessage(message string) ([]byte, error) {
+	return nil, ErrIsMasquerading
+}
+
+// Reloads wallet from disk
+func (w *masqueradeWallet) Reload() error {
+	_, err := w.loadStore()
+	return err
+}
+
+// Load the wallet store from disk and decrypt it
+func (w *masqueradeWallet) loadStore() (bool, error) {
```

Comment:
hmm, does this ever get called? Seems like we should probably just return an error here- can't imagine why we'd want to read the wallet store while masquerading.

Reply:
Sounds like a reasonable vector to simplify this a bit, right?